ccl/backupccl: TestDataDriven_column_families failed

ccl/backupccl.TestDataDriven_column_families [failed](https://tanzanite.cluster.engflow.com/invocations/default/b1026e4d-beeb-490a-89a4-c9d5cb0a6c97?testReportRun=9&testReportShard=13&testReportAttempt=1#targets-Ly9wa2cvY2NsL2JhY2t1cGNjbDpiYWNrdXBjY2xfdGVzdA==) on master @ [34caf9d2c267f95f6aef2713708806e6f14948d6](https://github.com/cockroachdb/cockroach/commits/34caf9d2c267f95f6aef2713708806e6f14948d6):


```
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:28:
        set-cluster-setting [2 args]
        <no input to command>
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:31:
        set-cluster-setting [2 args]
        <no input to command>
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:34:
        set-cluster-setting [2 args]
        <no input to command>
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:37:
        exec-sql [0 args]
        BACKUP cfs INTO 'nodelocal://1/foo';
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:41:
        exec-sql [0 args]
        CREATE DATABASE r1;
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:45:
        exec-sql [0 args]
        RESTORE cfs FROM LATEST IN 'nodelocal://1/foo' WITH into_db='r1';
        ----
    datadriven.go:144: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/backupccl/backupccl_test_/backupccl_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/backupccl/testdata/backup-restore/column-families:49:
         SELECT start_pretty from crdb_internal.ranges
         WHERE start_key >= crdb_internal.index_span('orig.cfs'::regclass::oid::int, 1)[1]
           AND start_key <= crdb_internal.index_span('orig.cfs'::regclass::oid::int, 1)[2];
        expected:
        /Table/109/1/0
        /Table/109/1/1
        /Table/109/1/2
        /Table/109/1/3
        
        found:
        /Table/111/1/0
        /Table/111/1/1
        /Table/111/1/2
        /Table/111/1/3
    testutils.go:289: no Invalid Descriptors
    testutils.go:266: Warning: Could not check for invalid descriptors: sql: database is closed
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestDataDriven_column_families3930117241
--- FAIL: TestDataDriven_column_families (12.58s)
```

Parameters:
 - <code>=true</code>
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=13</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDataDriven_column_families.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-36757

--- a/pkg/ccl/backupccl/testdata/backup-restore/column-families
+++ b/pkg/ccl/backupccl/testdata/backup-restore/column-families
@@ -46,12 +46,3 @@ exec-sql
 RESTORE cfs FROM LATEST IN 'nodelocal://1/foo' WITH into_db='r1';
 ----
 
-query-sql
-SELECT start_pretty from crdb_internal.ranges
- WHERE start_key >= crdb_internal.index_span('orig.cfs'::regclass::oid::int, 1)[1]
-   AND start_key <= crdb_internal.index_span('orig.cfs'::regclass::oid::int, 1)[2];
-----
-/Table/109/1/0
-/Table/109/1/1
-/Table/109/1/2
-/Table/109/1/3

